### PR TITLE
Testing/ak/books #35

### LINKED ISSUE #35: Testing: check to see a searched for book is rendered on page when found
- [ ] Books component renders this

### PR DESCRIPTION
**Pull request by amykarnaze**
#### What's this PR do?
-  Adds one test to check a book is rendered on the page with book information and that the reading list button can be fired.

Type of change made:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features

#### Where should the reviewer start?
- Book.test.js

#### Why is this change required? What problem does it solve?
- Provide 2 unit test to book component

#### Were there any challenges that arose while implementing this feature? If so, how were they addressed?
- isbn is not working

#### What are the relevant tickets?
- closes #35 

#### Questions:
- Any thought why the isbn is not working?

### CODE REVIEW

File: src/Components/Book/Book.test.js [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+    const bookAuthor = screen.getByText("Louise Penny")
+    const addBookButton = screen.getByRole("button", {name: "Reading List"})
+
+    expect(bookRank).toBeInTheDocument();
+    // expect(bookIsbn).toBeInTheDocument();
+    expect(bookTitle).toBeInTheDocument();
+    expect(bookImg).toBeInTheDocument();
+    expect(bookAuthor).toBeInTheDocument();
+    expect(addBookButton).toBeInTheDocument();
+  })
```
Comment (kip-west) [inline src/Components/Book/Book.test.js:42]: These unit tests look great!!

File: src/Components/Book/Book.test.js [hunk 2]
```diff
@@ -1,3 +1,65 @@
...
+    render(
+      <Provider store={store}>
+        <MemoryRouter>
+          <Book book={book1}/>
+        </MemoryRouter>
+      </Provider>
+    )
+    
+    const addBookButton = screen.getByRole("button", {name: "Reading List"})
+    fireEvent.click(addBookButton)
```
Comment (kip-west) [inline src/Components/Book/Book.test.js:63]: Great work getting this one going!!

I wonder if we should mock out little more here before making an assertion! I'm imagining something like...
`
// mock out this.handleClick() that's being passed down as addBook from Books Component
const mockAddBook = jest.fn()
// give this mocked function to our Book
<Book book={book1} addBook={mockAddBook}
// assert that our mocked function is called when the button is clicked
expect(mockAddBook).toBeCalled()
`

File: src/Components/Book/Book.test.js [hunk 3]
```diff
@@ -1,3 +1,65 @@
...
+
+    render(
+      <Provider store={store}>
+        <MemoryRouter>
+          <Book book={book1}/>
+        </MemoryRouter>
+      </Provider>
+    )
+
+    const bookRank = screen.getByText("1", { exact: false })
```
Comment (NakiNorton) [inline src/Components/Book/Book.test.js:29]: Nice work using { exact: false } for the rank! I'll have to add this into one of my tests as I couldn't work out how to find the element!

File: src/Components/Book/Book.test.js (hunk 2 above)
Comment (NakiNorton) [reply src/Components/Book/Book.test.js:63]: I like that idea Jake! It'll be an easy change and make the test a bit more robust.

**Review by NakiNorton — APPROVED** [review]

**Pull request merged**